packages/xo-server/src/xo-mixins/resource-sets.js

```diff
@@ -317,15 +317,15 @@ export default class {
     const sets = keyBy(await this.getAllResourceSets(), 'id')
     forEach(sets, ({ limits }) => {
       forEach(limits, (limit, id) => {
-        if (VM_RESOURCES[id]) {
+        if (VM_RESOURCES[id] || id.startsWith('ipPool:')) {
           // only reset VMs related limits
           limit.available = limit.total
         }
       })
     })
 
-    forEach(this._xo.getAllXapis(), xapi => {
-      forEach(xapi.objects.all, object => {
+    for (const xapi of Object.values(this._xo.getAllXapis())) {
```

Comment:
It's an unrelated change, you can do it in an other commit/PR.

Reply:
What's unrelated, exactly?

Reply:
The usage of `for(const ... of ...)` instead of the `forEach`

Reply:
It was related to the commit changes and necessary to make the async calls that compute the IP pools usages. But now I changed it to `Promise.all`s and `map`s in order to parallelize the calls.